plugins/include/open62541/plugin/pubsub_ethernet_xdp.h

```diff
@@ -0,0 +1,19 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information.
+ *
+ *   Copyright (c) 2019-2020 Kalycito Infotech Private Limited
+ */
+
+#ifndef UA_NETWORK_PUBSUB_XDP_ETHERNET_H_
+#define UA_NETWORK_PUBSUB_XDP_ETHERNET_H_
+
+#include <open62541/plugin/pubsub.h>
+
+_UA_BEGIN_DECLS
+
+UA_PubSubTransportLayer UA_EXPORT
+UA_PubSubTransportLayerEthernetXDP(void);

```

Comment:
I'm surprised you don't give more configuration options as input to the layer.
All the global defines in `plugins/ua_pubsub_ethernet_xdp.c` ought to be config options that are set at startup.

Reply:
The global defines are moved into a separate config structure.

The remaining defines shall be removed once XDP is mainlined to the stable linux kernel.